src/app/log_window.rs

```diff
@@ -22,67 +22,71 @@
 // terms of the Steamworks API by Valve Corporation, the licensors of this
 // Program grant you additional permission to convey the resulting work.
 
-use std::collections::VecDeque;
+use std::sync::mpsc::{Receiver, Sender};
 
-static BUFFER_CAPACITY: usize = 1 << 24;
+use ringbuf::{ring_buffer::RbBase, Rb};
 
 pub struct LogWindow {
     pub(super) term_shown: bool,
-    term: luminol_term::Terminal,
+    byte_rx: Receiver<u8>,
+    byte_tx: Sender<u8>,
+    ringbuf: RingBuf, // Should we use a temporary file instead of a ringbuffer?
```

Comment:
What can a `ringbuf::ring_buffer::LocalRb` do that a `std::collections::VecDeque` can't?

By the way, a temporary file can't be easily rotated when full in the same way that a ring buffer can. Having an unbounded buffer for logs isn't very safe.

Reply:
I thought it might be more optimized for our usecase (it'd perform overwriting writes directly to values rather than popping elements) but now that I look at the code it's actually popping a value then appending it. If we copy slices though, it does directly copy. I was thinking of optimizing the log to send byte slices anyway, so it might be worth keeping ringbuf.

Reply:
Actually, I'll just replace it with a vecdeque. The crate doesn't really do what I expected. `circular-buffer` would be better but it doesn't support runtime-sized buffers.